test/terraform_aws_s3_private_bucket_test.go

```diff
@@ -373,3 +373,78 @@ func TestTerraformAwsS3PrivateBucketNoLoggingBucket(t *testing.T) {
 	AssertS3BucketLoggingNotEnabled(t, awsRegion, testName)
 	AssertS3BucketPolicyContainsNonTLSDeny(t, awsRegion, testName)
 }
+
+func AssertS3BucketAnalyticsEnabled(t *testing.T, region string, bucketName string) {
+	err := AssertS3BucketAnalyticsEnabledE(t, region, bucketName)
+	require.NoError(t, err)
+}
+
+func AssertS3BucketAnalyticsEnabledE(t *testing.T, region string, bucketName string) error {
+	s3Client, err := aws.NewS3ClientE(t, region)
+
+	if err != nil {
+		return err
+	}
+
+	params := &s3.ListBucketAnalyticsConfigurationsInput{
+		Bucket: awssdk.String(bucketName),
+	}
+
+	bucketAnalytics, err := s3Client.ListBucketAnalyticsConfigurations(params)
+
+	if err != nil {
+		return err
+	}
+
+	analyticsEnabled := bucketAnalytics.IsTruncated
+
+	if analyticsEnabled == nil {
+		return fmt.Errorf("Analytics is not enabled")
+	}
+
+	return nil
+}
+
+func TestTerraformAwsS3PrivateBucketAnalyticsBucket(t *testing.T) {
+	t.Parallel()
+
+	// TODO: create & hook tests below to tempTestFolder /examples/analytics after logging module connected
+
+	tempTestFolder := test_structure.CopyTerraformFolderToTemp(t, "../", "examples/simple-no-logging")
```

Comment:
I don't see where you are creating a logging bucket to store the analytics data.

Reply:
That's what the todo in the comment on line 411 is about. I think you're right that

>there will need to be bucket policy changes needed on the aws-logs modules to support his change.

Because I can't get this test to pass when I write it correctly (or "correctly" afaict as a n00b).